includes/Admin/Admin_AJAX.php

```diff
@@ -43,24 +47,111 @@ public function get_nonce() {
 	}
 
 	/**
-	 * Runs checks.
+	 * Handles the AJAX request that returns the checks to run.
 	 *
 	 * @since n.e.x.t
 	 */
-	public function run_checks() {
-		$nonce = filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING );
+	public function get_checks_to_run() {
+		// Verify the nonce before continuing.
+		$this->verify_nonce();
 
-		if ( ! wp_verify_nonce( $nonce, self::NONCE_KEY ) ) {
+		$checks = filter_input( INPUT_POST, 'checks', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
+		$plugin = filter_input( INPUT_POST, 'plugin', FILTER_SANITIZE_STRING );
+
+		// Attempt to get the plugin basename based on the request.
+		try {
+			$plugin_basename = Plugin_Request_Utility::get_plugin_basename_from_input( $plugin );
+		} catch ( Exception $error ) {
 			wp_send_json_error(
-				new WP_Error( 'invalid-nonce', __( 'Invalid nonce', 'plugin-check' ) ),
+				new WP_Error( 'invalid-plugin', $error->getMessage() ),
 				403
 			);
 		}
 
+		$plugin_active = is_plugin_active( $plugin_basename );
+
+		// Create the checks instance.
+		$checks_instance = new Checks( WP_PLUGIN_DIR . '/' . $plugin_basename );
+		$all_checks      = $checks_instance->get_checks();
+
+		// Filter checks to run based on the request.
+		$checks_to_run = empty( $checks ) ? $all_checks : array_intersect_key( $all_checks, array_flip( $checks ) );
+
+		// If the plugin is inactive and there are runtime checks.
+		if ( ! $plugin_active && $this->has_runtime_check( $checks_to_run ) ) {
+			// If specific checks were requested, return an error.
+			if ( ! empty( $checks ) ) {
+				wp_send_json_error(
+					new WP_Error(
+						__( 'Runtime checks cannot be run against inactive plugins.', 'plugin-check' )
+					)
+				);
+			}
+
+			// If all checks are requested, filter out any runtime checks.
+			$checks_to_run = array_filter(
+				$checks_to_run,
+				function ( $check ) {
+					return ! $check instanceof Runtime_Check;
+				}
+			);
+		}
+
+		wp_send_json_success(
+			array(
+				'plugin' => $plugin,
+				'checks' => array_keys( $checks_to_run ),
+			)
+		);
+	}
+
+	/**
+	 * Run checks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function run_checks() {
+		// Verify the nonce before continuing.
+		$this->verify_nonce();
+
 		wp_send_json_success(
 			array(
 				'message' => __( 'Verified!', 'plugin-check' ),
 			)
 		);
 	}
+
+	/**
+	 * Verify the nonce passed in the request.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function verify_nonce() {
+		$nonce = filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING );
```

Comment:
Similarly here, this is directly coupled to the AJAX request, so it would be better to keep it in the individual AJAX callback methods. You could pass `$nonce` as a parameter to this method to keep it decoupled.